examples/heartbeat_demo/heart_monitor.py

```diff
@@ -0,0 +1,54 @@
+import random
+import time
+import pickle
+
+from nucypher.data_sources import DataSource
+
+
+HEART_DATA_FILENAME = 'heart_data.pickle'
+
+# Alicia defined a label to categorize all her heart-related data ❤️
+# All DataSources that produce this type of data will use this label.
+DEFAULT_LABEL = b"Alicia's heart data"
+
+
+def generate_heart_rate_samples(policy_pubkey,
+                                label: bytes = DEFAULT_LABEL,
+                                samples: int = 30,
+                                save_as_file: bool = False):
+    data_source = DataSource(policy_pubkey_enc=policy_pubkey,
+                             label=label)
+
+    data_source_public_key = bytes(data_source.stamp)
+
+    heart_rate = 80
+    now = time.time()
+
+    kits = list()
+    for _ in range(samples):
+        # Simulated heart rate data
+        # Normal resting heart rate for adults: between 60 to 100 BPM
+        heart_rate = random.randint(max(60, heart_rate-5),
+                                    min(100, heart_rate+5))
+        now += 3
+
+        heart_rate_data = {
+            'heart_rate': heart_rate,
+            'timestamp': now,
+        }
+
+        plaintext = pickle.dumps(heart_rate_data)
+        message_kit, _signature = data_source.encapsulate_single_message(plaintext)
+
+        kit_bytes = message_kit.to_bytes()
+        kits.append(kit_bytes)
+
+    data = {
+        'data_source': data_source_public_key,
+        'kits': kits,
+    }
+
+    if save_as_file:
+        pickle.dump(data, open(HEART_DATA_FILENAME, "wb"))
```

Comment:
Why do we dump with pickle btw?
(It may leave the wrong impression that ciphertexts should be dumped with pickle)